workbench-core/accounts/src/services/projectService.ts

```diff
@@ -254,6 +255,55 @@ export default class ProjectService {
     return newProject;
   }
 
+  /**
+   * Update the name or description of an existing project.
+   *
+   * @param request - a {@link UpdateProjectRequest} object that contains the id of the project to update
+   *                  as well as the new field values
+   * @returns a {@link Project} object that reflects the changes requested
+   */
+  public async updateProject(request: UpdateProjectRequest): Promise<Project> {
+    const { projectId, updatedValues } = request;
+
+    // disregard empty strings as valid input
+    if (updatedValues.name !== undefined && _.isEmpty(updatedValues.name)) {
+      updatedValues.name = undefined;
+    }
+    if (updatedValues.description !== undefined && _.isEmpty(updatedValues.description)) {
+      updatedValues.description = undefined;
+    }
+
+    // verify at least one attribute is being updated
+    if (!updatedValues.name && !updatedValues.description) {
+      throw Boom.badRequest('You must supply a new nonempty name and/or description to update the project.');
```

Comment:
I see that we're doing it elsewhere, but the service layer shouldn't be throwing network errors. This is where you'd throw an error from the service and do a try/catch in the controller to handle the response we want to send back to the network.

Reply:
Is this something I should fix now or can I leave it as a room for improvement item since this is consistent with most of the codebase and this service in particular? Happy to sync about refactoring the file as a whole but to my knowledge, this is what we are mostly doing in the services. I do see your point though, but my PR might not be the avenue for the fix.

Reply:
Going to refactor for all of project service before merging to develop